packages/foreman/foreman/foreman.spec

```
@@ -1005,18 +1005,20 @@ fi
 exit 0
 
 %post service
-%systemd_post %{name}.socket
 %systemd_post %{name}.service
 
 %preun service
-%systemd_preun %{name}.socket
 %systemd_preun %{name}.service
+%systemd_preun %{name}.socket
```

Like with `%post` I think you can combine them which means systemd can properly figure out the dependencies and ensure they're both stopped.
```suggestion
%systemd_preun %{name}.service %{name}.socket
```